geom/alg_intersection.go

```diff
@@ -2,16 +2,34 @@ package geom
 
 import "github.com/peterstace/simplefeatures/rtree"
 
-func intersectionOfLineStringAndLineString(
-	ls1, ls2 LineString,
-) (MultiPoint, MultiLineString) {
-	return intersectionOfLines(ls1.asLines(), ls2.asLines())
-}
-
-func intersectionOfMultiLineStringAndMultiLineString(
-	mls1, mls2 MultiLineString,
-) (MultiPoint, MultiLineString) {
-	return intersectionOfLines(mls1.asLines(), mls2.asLines())
+func intersectionOfIndexedLines(
+	lines1 indexedLines, lines2 indexedLines,
+) (
+	MultiPoint, MultiLineString,
+) {
+	// TODO: Investigate potential speed up of swapping lines.
```

Comment:
I'll create an additional ticket for this. To work out if it's worthwhile, we'll need to add some benchmarks specifically around this.